dora/core/client/fs/src/test/java/alluxio/client/file/cache/LocalCacheFileInStreamTest.java

```diff
@@ -568,6 +568,18 @@ public void testUnbuffer() throws Exception {
     Assert.assertEquals(1, manager.mPagesServed);
   }
 
+  @Test
+  public void testPositionReadFallBack() throws Exception
+  {
+    int pages = 10;
+    int fileSize = mPageSize * pages;
+    byte[] testData = BufferUtils.getIncreasingByteArray(fileSize);
+    ByteArrayCacheManager manager = new ByteArrayCacheManager();
+    LocalCacheFileInStream stream = setupWithSingleFile(testData, manager);
+
+    Assert.assertEquals(100, stream.positionedRead(0, new byte[10], 100, 100));
```

Comment:
Is the above fallback mainly for the case when the passed in byte array is of small size compared to off/len ? If so, can we sidestep the fallback and just check for the buff length etc (using an if condition instead of a try / catch)  and directly go to the fallback codepath ?

Reply:
I'm not sure if there is any other fallback case in your production environment, so I'm just guessing and mimic if there is a smaller size array passed in, then we can fallback to the external input stream directly.